### PR TITLE
Upgrade to v4

### PR DESCRIPTION
**Pull request by borisrorsvort**
Tedious upgrade towards middleman v4. What's in:

- No imageoptim anymore (did not manage to make it work)
- New deployment method (much cleaner imho)
- Managed to keep sprockets though advise to switch for gulp like asset management (https://middlemanapp.com/advanced/asset_pipeline/)
- Adds directory indexes so that you dont need to make a folder +index for clean urls
- upgraded ruby version

### CODE REVIEW

**Comment by borisrorsvort** [comment]
@toch @otagi Could you have a look?

**Comment by borisrorsvort** [comment]
@toch @otagi It's been a while now. Could you have a look?

File: README.md [hunk 1]
```diff
@@ -16,9 +16,11 @@ The site is based on [Middleman](https://middlemanapp.com) and the [Agency](http
 
 To deploy the master branch on Github pages:
 
+Make sure you have the `mgd` gem installed [docs](https://github.com/hovancik/middleman-github-deploy) and pushed all changes to github.
```
Comment (toch) [inline README.md]: Why it's no in the Gemfile?

File: config.rb [hunk 2]
```diff
@@ -41,23 +46,9 @@ def ruby_shops
...
   activate :minify_html
   activate :gzip
-  activate :imageoptim
   activate :asset_hash
-
-  require_relative "./lib/build_cleaner"
-  activate :build_cleaner
-end
-
-activate :google_analytics do |ga|
```
Comment (toch) [inline config.rb]: We were using a plugin for that before. If you remove it, I guess it's because it's no more available for the v4?

**Comment by toch** [comment]
@borisrorsvort sorry for the delay, done :) and thanks :clap:

File: config.rb (hunk 2 above)
Comment (borisrorsvort) [inline config.rb]: Indeed, it has not been updated for v4 yet.

File: source/assets/stylesheets/all.css.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
+@charset "UTF-8";
+@import url(https://fonts.googleapis.com/css?family=Raleway:300,400);
```
Comment (flexbox) [inline source/assets/stylesheets/all.css.scss]: You can drop this and add directly on the main layout with `= stylesheet_link_tag 'https://fonts.googleapis.com/css?family=Raleway:300,400'` #performanceMatters 😄

**Comment by borisrorsvort** [comment]
@otagi Can you check, merge and deploy plz ?

**Pull request merged**